test/should.js

```diff
@@ -688,44 +688,33 @@ describe('should', function() {
     'test'.should.have.property('length');
     (4).should.not.have.property('length');
     ({ 1: 1 }).should.have.property(1);
+    ({ a: 1 }).should.have.property('toString');
 
     err(function(){
       'asd'.should.have.property('foo');
-    }, "expected 'asd' to have a property 'foo'");
-  });
-
-  it('nested.property(name)', function(){
-    ({ 'foo.bar': 'baz'}).should.not.have.nested.property('foo.bar');
-    ({ foo: { bar: 'baz' } }).should.have.nested.property('foo.bar');
-
-    ({ 'foo': [1, 2, 3] }).should.have.nested.property('foo[1]');
-
-    ({ 'foo.bar[]': 'baz'}).should.have.nested.property('foo\\.bar\\[\\]');
-
-    err(function(){
-      ({ 'foo.bar': 'baz' }).should.have.nested.property('foo.bar');
-    }, "expected { 'foo.bar': 'baz' } to have a nested property 'foo.bar'");
+    }, "expected 'asd' to have property 'foo'");
   });
 
   it('property(name, val)', function(){
```

Comment:
IMO it would be good to have tests to see if we're really unable to use both `own` and `nested` together.